indexd/index/drivers/alchemy.py

```diff
@@ -416,11 +418,24 @@ def update(self, did, rev, urls=None, file_name=None, version=None):
                 raise RevisionMismatch('revision mismatch')
 
             if urls is not None:
+                for url in record.urls:
+                    session.delete(url)
+
                 record.urls = [IndexRecordUrl(
                     did=record.did,
                     url=url
                 ) for url in urls]
 
+            if metadata is not None:
+                for md_record in record.index_metadata:
+                    session.delete(md_record)
```

Comment:
I feel like deleting all entries and creating new one with only what was supplied might lead to unexpected outcomes. For example say, I have an entry with a UUID stored somewhere else like in another db, having multiple meta data records. I might later want to update or set new field in the metadata without doing an indexd look up first. I would expect this to append to the existing entries instead of removing previous entries and leaving just this one. 
A way to do this will be to create a dict of the existing metadata entries before deleting and update the dict based on what was supplied before creating the metadata entries

Reply:
But then you have no way to delete/fix previous metadata record if the key is wrong, so I think updating the whole thing should be the first thing to support. If we want to support partial update, that should be a PATCH instead of PUT